chainer/variable.py

```diff
@@ -724,6 +724,16 @@ def __len__(self):
         """
         return len(self.array)
 
+    def __iter__(self):
+        if chainer.config.enable_backprop:
+            warnings.warn(
+                'Backprop through `Variable.__iter__` is slow. Consider'
+                ' using `chainer.functions.split_axis`.',
+                RuntimeWarning)
```

Comment:
Chainer could add `PerformanceWarning` like cupy/cupy#1607.

Reply:
I think it is better, as there are some other functions that may want it (e.g. `chainer/functions/math/erf.py`)